crates/repository/src/cargo.rs

```diff
@@ -0,0 +1,187 @@
+    host: Host,
+    environment: Environment,
+    arguments: Vec<OsString>,
+}
+
+impl Cargo {
+    pub fn local(environment: Environment) -> Self {
+        Self {
+            host: Host::Local,
+            environment,
+            arguments: Vec::new(),
+        }
+    }
+
+    pub fn remote(environment: Environment) -> Self {
+        Self {
+            host: Host::Remote,
+            environment,
+            arguments: Vec::new(),
+        }
+    }
+
+    pub async fn setup(&self, repository: &Repository) -> Result<()> {
+        if let Environment::Sdk { version } = &self.environment {
+            match self.host {
+                Host::Local => {
+                    let data_home = get_data_home().wrap_err("failed to get data home")?;
+
+                    download_and_install(version, data_home)
+                        .await
+                        .wrap_err("failed to download and install SDK")?;
+                }
+                Host::Remote => {
+                    let mut command = Command::new(repository.root.join("scripts/remoteWorkspace"));
+
+                    let status = command
+                        .arg("pepsi")
+                        .arg("sdk")
+                        .arg("install")
+                        .arg("--version")
+                        .arg(version)
+                        .status()
+                        .await
+                        .wrap_err("failed to run pepsi")?;
+
+                    if !status.success() {
+                        bail!("pepsi failed with {status}");
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn arg(&mut self, argument: impl Into<OsString>) -> &mut Self {
+        self.arguments.push(argument.into());
+        self
+    }
+
+    pub fn args(&mut self, arguments: impl IntoIterator<Item = impl Into<OsString>>) -> &mut Self {
+        self.arguments.extend(arguments.into_iter().map(Into::into));
+        self
+    }
+
+    pub fn command(
+        self,
+        repository: &Repository,
+        compiler_artifacts: &[impl AsRef<Path>],
+    ) -> Result<Command> {
+        let arguments = self.arguments.join(OsStr::new(" "));
+
+        let relative_pwd = diff_paths(
+            current_dir().wrap_err("failed to get current directory")?,
+            &repository.root,
+        )
+        .wrap_err("failed to express current directory relative to repository root")?;
+
+        let command_string = match self.environment {
+            Environment::Native => {
+                let mut command = OsString::from("cargo ");
+                command.push(arguments);
+                command
+            }
+            Environment::Sdk { version } => {
+                let data_home = get_data_home().wrap_err("failed to get data home")?;
+                let environment_file = &data_home.join(format!(
+                    "sdk/{version}/environment-setup-corei7-64-aldebaran-linux"
+                ));
```

Comment:
This goes kaputt when building remotely. We have to use the data home on the remote which we'd either need to retrieve first or use shell magic to implement the logic from `get_data_home` right inside the command we're building here.

Reply:
uff... What way do you suggest to fix that?